### ISSUE TITLE
Select work type for new work or new batch of works

### ISSUE THREAD

**Issue by hannahfrost** [projecthydra-labs/hyku#580]
As a user creating a new work or a new batch of works, when I make my selection on the Works menu in the top menu bar, I should be directed to a page where I can indicate the type of work (generic, image, etc.)

**Comment by ggeisler** [comment]
@hannahfrost: proposed mockup below.

**1. User selects "New Work" from the Works menu.** Note that I suggest that menu item be re-labeled to "Create new work" but maybe that is done in another ticket.

**2. The Select Type of Work page is displayed as shown below.** This page should be displayed in the Works section of the administration side of the app, although it doesn't appear that we are showing the works-related pages in that context yet.

![select-work](https://cloud.githubusercontent.com/assets/101482/21702411/71367bd6-d368-11e6-8df8-b3a997070092.png)

The user either accepts the default value of generic work, or selects another available work type. Clicking the Create work button leads to the existing Create New Work page, although I suggest we change the title of that page to "Describe New Work", given this less streamlined workflow. (Technically the work might not be created until the user saves it on the Describe New Work page, but the user doesn't know that, and I think it might make more sense to the user that the flow is 1. Select work type, 2. Create work, and 3. Describe that work.)

![describe-work](https://cloud.githubusercontent.com/assets/101482/21702420/7b5cc6a6-d368-11e6-83ca-a618875eb7a2.png)

**3. If the user clicks the Cancel button on the Select Type of Work page**, where should they end up?

As the app is now, they can select the New Work menu item from anywhere in the app, because the Works menu is shown in the top/brand bar and is visible everywhere. However, I propose in https://github.com/projecthydra-labs/hyku/issues/502 to de-clutter the top/brand bar and thus eventually I'd expect that the user will only select the New Work action from either the homepage or the Works page in the Admin side of the app. 

So depending on whether we want to take a short- or long-term view, we can either return the user to whatever page they were on when they choose the New Work menu item (short-term view), or (long-term view) just put them on the Works index page (which should eventually have a Create New Work button on it, similar to the Admin Set index page, so they can easily attempt again if they didn't really intend to cancel). I'd recommend the second approach of redirecting to the Works index page but that assumes we will improve the Works index page and add a Create new work button to it and de-clutter the top/brand bar as proposed in #502.

**Comment by hannahfrost** [comment]
@ggeisler I like all of this. I agree that it would be good to take the long-term view for the Works index page. It is time to bring it in alignment with what we are doing with Admin Sets in that regard. 

This could be a discussion item at stand-up, though unfortunately I have a meeting at 10:30, so my availability is limited. In any event, I'm going to move this ticket to the ready column.

**Comment by mjgiarlo** [comment]
Thoughts on the above, @jcoyne?

**Comment by jcoyne** [comment]
I just am no clear on why this should be a form with radio buttons instead of a list of links.  The latter would enable you to do with one click, which would require two clicks with the approach specified. (The radio form also requires us to write more code, so I may just be lazy)

**Comment by ggeisler** [comment]
Because from the user point of view this is the first step of a _multi-step process_, not just navigating to a new page as links are commonly used.

The submit and close buttons are there to give the user the opportunity to make a deliberate choice of work type or to cancel the process (if the choices are links, there is no purpose in having a submit button, but you still need a cancel action in case the user was just curious about the feature or otherwise doesn't actually want to create a new work). The descriptions of the work types could be long, and eventually the number of work types could expand, making the list longer and potentially feel more complicated. The goal for this step is to begin a process where the user understands what's going on and won't accidentally make a choice that wasn't intended. The use of a form with radio buttons supports that goal much better than a list of links, and is worth the extra click.

If that doesn't convince you I'm not sure what else to say. Both of the UX designers at Stanford liked the original, modal design (I reviewed the original design with Jennifer before posting it) and none of the sufia-ux people who were tagged objected to the design. The original modal version is very similar to existing implementations elsewhere (e.g., when you create a child page on the DuraSpace wiki, you select the type of page in modal and have to confirm the choice with a submit button) so it's not like this is a novel or crazy approach. 

I wasn't included in the discussion with @hannahfrost and @mjgiarlo where it was decided that the original modal design wasn't worth the development effort, but if the development effort objected to there was the same objection expressed here of not wanting to write more code to support the use radio buttons versus links, I think that was the wrong decision, because the modal approach would probably be a better experience than what's mocked up in this ticket, though the mockup here is still better for the user than a list of links.

**Comment by hannahfrost** [comment]
@ggeisler - @jcoyne and @mjgiarlo and I were working on ticketing some things today and revisited this issue again. We are going to proceed with the radio button selection in a modal as originally designed. It will be used for both creation of batch works or singleton works.

**Comment by atz** [comment]
My official policy is that I don't care to argue much whatever you decide to design.  

That said, personally, I prefer to have the buttons actualized in the modal.  The destination is not harmful or particularly costly to hit.  It is, in fact, just a GET request and should not require any additional protection or confirmation.  

The mockup seems to have excluded icons.  Please clarify whether this is intentional.  Including icons in the most conventional (current) place, namely left-most, will add separation between the radio circle and the text, and introduce variability when some types have an icon and others do not. 

To say that the User views this as the first in a multi-step process is not necessarily universal.  A given user could be interested in only one type for a given period.  For them the selector is "in the way" of what they perceive as the first step where they feel they actually do something.  They could as easily bookmark the `.../new` page (w/ parameter) and start their shift there.  The actionable output of the selector form is just a link, technically, and it is not fundamentally important UX-wise to hide that from the user.  

Compare:
- What is the user getting when they click a submit button having selected a **radio option**?  They don't actually know.  Even when the action is a GET, the browser doesn't tell them beforehand.  
- What is the user getting when they click a **link (or link button)**?  They know exactly because the browser tells them on hover-over.  

Coding the modal stuff was only a pain because the hidden problem of multiple types broke other parts of the app just by existing.  It was not because the modal itself was doing anything special or uniquely difficult.  

Basically, I recommend we take another look after my PR is merged (which is complete and green locally, w/ one fiddly hangup in Travis).  

My reservation about the modal approach is scalability, specifically when an institution has dozens of types.  Then we would need to get into pagination, auto-completion, etc.  Notably, having radio buttons in that case only makes it worse.  When a radio is only presenting a "page" of options, the UX metaphor is broken.

**Comment by mjgiarlo** [comment]
@atz @ggeisler @hannahfrost @jcoyne Let's discuss after stand-up today. I'll kick it off after I adjourn the meeting for the others on the call.

**Comment by ggeisler** [comment]
Responses to comments from @atz:

> That said, personally, I prefer to have the buttons actualized in the modal. The destination is not harmful or particularly costly to hit. It is, in fact, just a GET request and should not require any additional protection or confirmation.

Not sure what you mean by "buttons actualized in the modal" so I'm not sure if this is making an argument for something different than was mocked up. 

> The mockup seems to have excluded icons. Please clarify whether this is intentional. Including icons in the most conventional (current) place, namely left-most, will add separation between the radio circle and the text, and introduce variability when some types have an icon and others do not.

There are icons in the modal mockup (see below). You must be referring to the non-modal mockup I was asked to do after the modal mockup. The non-modal does intentionally exclude icons, because it would be weird to have icons next to radio buttons. In the modal version, I would follow the fairly standard practice when using icons in this way of having a default icon that is used when there is not a custom icon, so there would not be a scenario where only some types have an icon and others do not.

![ba04db1c-a1e2-11e6-8d5e-1dbc5a72eb30](https://cloud.githubusercontent.com/assets/101482/22032130/60deff38-dca0-11e6-9e12-ee3427080d91.png)

> To say that the User views this as the first in a multi-step process is not necessarily universal. [...]

This feature was assigned to me specifically as "we need **an initial step** before the Add Work page where the user selects the work type they want to create, so that we can present the appropriate metadata form for the selected work type." The other things you're talking about here sound like you envision an advanced user or a developer who is working on the application. In reality, adding a new work is likely to be a feature used by many users infrequently or for the first time (e.g., self-deposit) and even for more frequent users, expecting them to look at and interpret the link address they see on hover is very optimistic. Normal users, the ones we should be designing for, neither think like nor have the technical knowledge that developers do.

> My reservation about the modal approach is scalability, specifically when an institution has dozens of types. Then we would need to get into pagination, auto-completion, etc. Notably, having radio buttons in that case only makes it worse. When a radio is only presenting a "page" of options, the UX metaphor is broken.

I might be wrong, but I don't think we foresee an institution having "dozens of types" of works. If that is more than an edge case, then sure, the modal approach might not be ideal. Obviously I wouldn't suggest a radio group span multiple page. But unless we really expect institutions to have more than 15-18 work types, we can split the work type choices into two columns--and if necessary add scrolling--and the modal could accommodate a reasonably large number of choices.

**Comment by mjgiarlo** [comment]
Recapping my discussion with @hannahfrost @ggeisler @jcoyne @atz today:

1. The design we're implementing is @ggeisler's original. It uses a modal interface, displaying the name, description, and an icon for each work type, and the icons are styled radio buttons. This modal is used for new single works and new batches of works whenever there are *n > 1* work types generated in the application. At the bottom of the modal there are two buttons, "Close" and "Create work" for single works and "Close" and "Create batch of works" for batches.
1. We do want an icon for each type of work, yes. The initial implementation can be simple radio buttons, and @ggeisler has offered to style those in a follow-up PR.
1. In the meantime, we are not prioritizing design of the modal when the number of work types exceeds five types. This work has been written up for a subsequent work cycle: https://github.com/projecthydra-labs/hyrax/issues/274
1. @jcoyne and @atz will work together to get projecthydra/sufia#2970 merged in its current form, and the above work will be based on this.
